src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,198 @@
+            dateFrequencyMap.put(new YearMonthDay("Error"), new Frequency(0));
+            return dateFrequencyMap;
+        }
+        
+        try {
+            for (int i = 0; i < expandedData.length; i += (NUM_YEAR_BYTES + NUM_FREQUENCY_BYTES)) {
+                byte[] encodedYear = new byte[] {expandedData[i], expandedData[i + 1], expandedData[i + 2], expandedData[i + 3]};
+                int decodedYear = Base256Compression.bytesToInteger(encodedYear);
+                log.debug("Deserialize decoded the year " + decodedYear);
+                // TODO Extra 4 bytes are being written out in serialize - need to figure this out and remove 2 lines below.
+                if (i == expandedData.length - NUM_BYTES_PER_FREQ_VALUE)
+                    break;
+                /*
+                 * Decode the frequencies for each day of the year.
+                 */
+                for (int j = NUM_YEAR_BYTES; j < DAYS_IN_LEAP_YEAR * NUM_BYTES_PER_FREQ_VALUE + NUM_YEAR_BYTES; j += NUM_BYTES_PER_FREQ_VALUE) {
+                    int k = i + j;
+                    byte[] encodedfrequencyOnDay = new byte[] {expandedData[k], expandedData[k + 1], expandedData[k + 2], expandedData[k + 3]};
+                    int decodedFrequencyOnDay = Base256Compression.bytesToInteger(encodedfrequencyOnDay);
+                    if (decodedFrequencyOnDay != 0) {
+                        dateFrequencyMap.put(new YearMonthDay(decodedYear + OrdinalDayOfYear.calculateMMDD(j / NUM_BYTES_PER_FREQ_VALUE, decodedYear)),
+                                        new Frequency(decodedFrequencyOnDay));
+                        log.debug("put key value pair in SimpleDateFrequency map: " + decodedYear
+                                        + OrdinalDayOfYear.calculateMMDD(j / NUM_BYTES_PER_FREQ_VALUE, decodedYear) + "-" + decodedFrequencyOnDay);
+                    }
+                    
+                }
+                
+            }
+        } catch (IndexOutOfBoundsException indexOutOfBoundsException) {
+            log.error("Error decoding the compressed array of date values. Expanded array length: " + expandedData.length, indexOutOfBoundsException);
+        }
+        
+        return dateFrequencyMap;
+    }
+    
+    /**
+     * This is a helper class that will compress the yyyyMMdd and the frequency date concatenated to it without a delimiter
+     */
+    public static class Base256Compression {
+        
+        public static byte[] numToBytes(long num) {
+            return new byte[] {(byte) (num >>> 24), (byte) (num >>> 16), (byte) (num >>> 8), (byte) num};
+        }
+        
+        public static void writeToOutputStream(long num, ByteArrayOutputStream baos) {
+            baos.write((byte) (num >>> 24));
+            baos.write((byte) (num >>> 16));
+            baos.write((byte) (num >>> 8));
+            baos.write((byte) (num));
+        }
+        
+        public static int bytesToInteger(byte[] byteArray) {
```

Comment:
Please remove this method.  You should at no time be allocating a 4 byte array for use with this method.